File: Factorial.m
%FACTORIAL
%TAREA 4
%MIGUEL ANTONIO ORELLANA RUÍZ
%201801279

clc
clear all
close all
inicio=0;
while inicio==0;
    disp(' ');
    disp(' ');
    disp('------CALCULADORA DE FACTORIAL-------');
    disp('1. Calcular el factorial de un número');
    disp('2. Salir');
    opcion = input('Introduzca la opción del menú a realizar : ');
 try
    switch opcion
        case 1, disp('--------------FACTORIAL--------------');
                num =input('Introduzca un número:');
                if num<0
                    fprintf('El factorial de números negativos no existe\n');
                elseif num==0
                fprintf('El factorial de 0 es 1\n');
                else
                factorial=1;
                for i=1:num
                    factorial=factorial*i; 
                end
                fprintf('El resultado del factorial es=: %d \n',factorial);
                end
                archivo = fopen('factorialmat.txt','at');
                fprintf(archivo,'El factorial de %g, es %g \n', num, factorial);
                fclose(archivo);

           case 2, back = input('¿Esta seguro que desea salir? s/n: ','s');
            if back=='n' %Sí
                inicio=0;
                    clc
            elseif back=='s' %No
            inicio=1;
            end
    end
 catch
        disp('Ingrese un valor correcto')
        Inicio = 1;
end
end

File: Factorial.py

def pedir_numero_menú():
    correcto = False
    num = 0
    while (not correcto):
        try:
            num = int(input("\n>Ingrese una opción del menú:"))
            print()
            correcto = True
        except ValueError:
            print("Error, Ingrese un número que corresponda a una opción del menú\n")
    return num

def pedir_numero():
    correcto = False
    num = 0
    while (not correcto):
        try:
            num = int(input("Ingrese número:"))
            if (num >= 0):
                print()
                correcto = True
            else:
                print("Los factoriales de números negativos están indefinidos. Ingrese un entero positivo")
            print()
        except ValueError:
            print("Error, Ingrese un número entero positivo.\n")
    return num

salir = False
opcion = 0
while (not salir):
    print('-------MENÚ-------')
    print("1. Ingresar número")
    print("2. Historial")
    print("3. salir")
    opcion = pedir_numero_menú()
    if opcion == 1:
        archivo = open("Prog1.txt", "a")
        fac = pedir_numero()
        factorial=1
        fac1=str(fac)
        archivo.write("El número ingresado es: " + fac1);
        archivo.write("\n");
        if (fac % 7 ==0):
            for n in range(fac):
                factorial = factorial * (n + 1)
            factorial1=str(factorial)
            print("El factorial es: ", factorial)
            print()
            archivo.write("El factorial es: 1"+factorial1);
            archivo.write("\n");
            archivo.write("\n")
        else:
            print("Error. Ingrese un número divisible entre 7.\n")
            archivo.write("No es un número divisible entre 7.");
            archivo.write("\n");
            archivo.write("\n")
        archivo.close()
    elif opcion == 2:
        archivo = open("Prog1.txt", "r")
        historial = archivo.read()
        archivo.close()
        print(historial)
    elif opcion == 3:
        salir = True
    else:
        print("Ingrese un número que corresponda a una opción del menú\n")

print('   ** Miguel Antonio Orellana Ruíz**')